packages/gel/src/baseConn.ts

```diff
@@ -537,12 +551,31 @@ export class BaseRawConnection {
       if (this.stateCodec === INVALID_CODEC || this.stateCodec === NULL_CODEC) {
         wb.writeInt32(0);
       } else {
-        if (!this.stateCache.has(state)) {
-          const buf = new WriteBuffer();
-          this.stateCodec.encode(buf, state._serialise(), NOOP_CODEC_CONTEXT);
-          this.stateCache.set(state, buf.unwrap());
+        let encodedState: Uint8Array | null = null;
+
+        // Handle explicit transaction options isolation level, which should be
+        // set as the default isolation level for this query if it is not already
+        // in a transaction. If they match, do nothing.
+        if (
+          state.transactionOptions.isolation !==
+          state.config.get("default_transaction_isolation")
+        ) {
+          let newState = state;
+          if (isExecute && !this.isInTransaction()) {
```

Comment:
Why is this `if (isExecute` important? Just curious, but this deserves a comment.

Reply:
I matched the logic from the Python code that only modifies the state if called via execute. I don't have a complete picture of the full logic in my head though, so maybe this should only apply to the auto-select case?